File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.java
import java.io.*;

public class Main {
    static int sum = 0;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String S = br.readLine();
        String[] str = S.split("-");
        sum += calculate(str[0]);
        for(int i = 1; i < str.length; i++) {
            sum -= calculate(str[i]);
        }
        System.out.println(sum);
    }

    static int calculate(String S) {
        String[] arr = S.split("\\+");
        int tmp = 0;
        for(int i = 0; i < arr.length; i++) {
            tmp += Integer.parseInt(arr[i]);
        }
        return tmp;
    }
}

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.js
function calculate(S) {
  const arr = S.split('+').map(Number);
  let tmp = 0;
  for(let num of arr) {
    tmp += num;
  }
  return tmp;
}
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('-');
let sum = 0;
sum += calculate(input[0]);
for(let i = 1; i < input.length; i++) {
  sum -= calculate(input[i]);
}
console.log(sum);